### PR TITLE
PYTHON-4631: Pushed PREPARE_SHELL creation into an env.sh file

### PR DESCRIPTION
**Pull request by Jibola**
As part of the larger epic to remove our dependency on in-lined scripts. We're making a conscious effort to remove all the shell.exec calls in our evergreen tests and make then subprocess calls that call into a file. The first step in that process comes from removing our biggest offender "PREPARE_SHELL".

This PR will achieve a few things: 
- [x] - Move the PREPARE_SHELL call to the `configure-expansions.sh` file
- [x] - Create a directory for "scripts" within evergreen to keep the code tidy

### CODE REVIEW

**Comment by blink1073** [comment]
You have a merge conflict.  🫤

File: .evergreen/config.yml [hunk 1]
```diff
@@ -236,7 +200,7 @@ functions:
     - command: archive.targz_pack
       params:
         target: "drivers-dir.tar.gz"
-        source_dir: ${DRIVERS_TOOLS}
```
Comment (blink1073) [inline .evergreen/config.yml:239]: Perhaps we should explicitly make DRIVERS_TOOLS and PROJECT_DIRECTORY into expansions?

Comment (Jibola) [reply .evergreen/config.yml:239]: I had hoped we could remove the expansions.update call entirely, but I agree here.

File: .evergreen/config.yml [hunk 2]
```diff
@@ -469,26 +433,28 @@ functions:
...
+          . .evergreen/scripts/env.sh
           bash ${DRIVERS_TOOLS}/.evergreen/auth_aws/setup_secrets.sh drivers/atlas_connect
           PROJECT_DIRECTORY="${PROJECT_DIRECTORY}" \
             PYTHON_BINARY="${PYTHON_BINARY}" \
             TEST_ATLAS=1 \
             bash ${PROJECT_DIRECTORY}/.evergreen/hatch.sh test:test-eg
 
   "get aws auth secrets":
-    - command: subprocess.exec
+    - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:450]: I don't think these aws tests need to be shell.exec anymore

File: .evergreen/config.yml [hunk 3]
```diff
@@ -535,21 +501,23 @@ functions:
...
-          ${PREPARE_SHELL}
+          . .evergreen/scripts/env.sh
           # Test with and without AWS_ROLE_SESSION_NAME set.
           .evergreen/run-mongodb-aws-test.sh web-identity
           AWS_ROLE_SESSION_NAME="test" \
             .evergreen/run-mongodb-aws-test.sh web-identity
 
   "run oidc auth test with test credentials":
-    - command: subprocess.exec
+    - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:511]: I don't think this needs to be shell.exec anymore

File: .evergreen/config.yml [hunk 4]
```diff
@@ -660,15 +628,16 @@ functions:
         role_arn: ${aws_test_secrets_role}
 
   "setup atlas":
-    - command: subprocess.exec
+    - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:631]: ditto

File: .evergreen/config.yml [hunk 5]
```diff
@@ -835,19 +804,21 @@ task_groups:
     setup_group:
       - func: "fetch source"
       - func: "prepare resources"
-      - command: subprocess.exec
+      - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:807]: ditto

File: .evergreen/config.yml [hunk 6]
```diff
@@ -889,11 +860,12 @@ task_groups:
           args:
             - ${DRIVERS_TOOLS}/.evergreen/csfle/azurekms/create-and-setup-vm.sh
     teardown_group:
-      - command: subprocess.exec
+      - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:863]: ditto

File: .evergreen/config.yml [hunk 7]
```diff
@@ -966,11 +938,12 @@ task_groups:
           args:
             - ${DRIVERS_TOOLS}/.evergreen/auth_oidc/setup.sh
     teardown_task:
-      - command: subprocess.exec
+      - command: shell.exec
```
Comment (blink1073) [inline .evergreen/config.yml:941]: .

**Review by blink1073 — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml [hunk 8]
```diff
@@ -547,7 +512,7 @@ functions:
       params:
         working_dir: "src"
         binary: bash
-        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
+        include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```
Comment (Jibola) [inline .evergreen/config.yml:515]:
```suggestion
        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```

File: .evergreen/config.yml [hunk 9]
```diff
@@ -1889,14 +1854,14 @@ tasks:
...
-            ${PREPARE_SHELL}
+            . src/.evergreen/scripts/env.sh
             cd src
             git add .
             git commit -m "add files"
             export AZUREOIDC_DRIVERS_TAR_FILE=/tmp/mongo-python-driver.tgz
             git archive -o $AZUREOIDC_DRIVERS_TAR_FILE HEAD
             export AZUREOIDC_TEST_CMD="OIDC_ENV=azure ./.evergreen/run-mongodb-oidc-test.sh"
-            bash $DRIVERS_TOOLS/.evergreen/auth_oidc/azure/run-driver-test.sh
+            bash ${DRIVERS_TOOLS}/.evergreen/auth_oidc/azure/run-driver-test.sh
```
Comment (Jibola) [inline .evergreen/config.yml:1864]:
```suggestion
            bash $DRIVERS_TOOLS/.evergreen/auth_oidc/azure/run-driver-test.sh
```

File: .evergreen/config.yml [hunk 10]
```diff
@@ -1916,7 +1881,7 @@ tasks:
             # Ensure that we source the environment file created for us, set up any other variables we need,
             # and then run our test suite on the vm.
             export GCPOIDC_TEST_CMD="OIDC_ENV=gcp ./.evergreen/run-mongodb-oidc-test.sh"
-            bash $DRIVERS_TOOLS/.evergreen/auth_oidc/gcp/run-driver-test.sh
+            bash ${DRIVERS_TOOLS}/.evergreen/auth_oidc/gcp/run-driver-test.sh
```
Comment (Jibola) [inline .evergreen/config.yml:1884]:
```suggestion
            bash $DRIVERS_TOOLS/.evergreen/auth_oidc/gcp/run-driver-test.sh
```

File: .evergreen/config.yml [hunk 11]
```diff
@@ -1985,8 +1949,8 @@ tasks:
       - command: subprocess.exec
         params:
           binary: bash
-          working_dir: src
           include_expansions_in_env: ["DRIVERS_TOOLS"]
+          working_dir: src
```
Comment (Jibola) [inline .evergreen/config.yml:1953]:
```suggestion
          working_dir: src
          include_expansions_in_env: ["DRIVERS_TOOLS"]
```

File: .evergreen/config.yml [hunk 12]
```diff
@@ -2051,10 +2015,10 @@ tasks:
             shell: "bash"
             working_dir: src
             script: |
-              ${PREPARE_SHELL}
+              . .evergreen/scripts/env.sh
               set -x
-              export CONFIG=$PROJECT_DIRECTORY/.github/reviewers.txt
-              export SCRIPT="$DRIVERS_TOOLS/.evergreen/github_app/assign-reviewer.sh"
+              export CONFIG=${PROJECT_DIRECTORY}/.github/reviewers.txt
+              export SCRIPT="${DRIVERS_TOOLS}/.evergreen/github_app/assign-reviewer.sh"
```
Comment (Jibola) [inline .evergreen/config.yml:2021]:
```suggestion
              export CONFIG=$PROJECT_DIRECTORY/.github/reviewers.txt
              export SCRIPT="$DRIVERS_TOOLS/.evergreen/github_app/assign-reviewer.sh"
```

File: .evergreen/config.yml [hunk 13]
```diff
@@ -88,14 +51,14 @@ functions:
     - command: shell.exec
       params:
         script: |
-          ${PREPARE_SHELL}
+          . src/.evergreen/scripts/env.sh
           set -o xtrace
-          rm -rf $DRIVERS_TOOLS
-          if [ "${project}" = "drivers-tools" ]; then
+          rm -rf ${DRIVERS_TOOLS}
```
Comment (Jibola) [inline .evergreen/config.yml:56]:
```suggestion
          rm -rf $DRIVERS_TOOLS
```

Comment (Jibola) [reply .evergreen/config.yml:56]: remove curly braces on $DRIVERS_TOOLS in prepare resources step

**Comment by blink1073** [comment]
> FAILED test/test_discovery_and_monitoring.py::TestUnifiedLoggingStandalone::test_Topology_lifecycle - AssertionError: 7 != 8

I opened [PYTHON-4705](https://jira.mongodb.org/browse/PYTHON-4705) for this.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by blink1073 — APPROVED** [review]

File: .evergreen/config.yml [hunk 14]
```diff
@@ -454,10 +417,11 @@ functions:
       type: test
       params:
         working_dir: "src"
-        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
+        include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```
Comment (Jibola) [inline .evergreen/config.yml:420]:
```suggestion
        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```

File: .evergreen/config.yml [hunk 15]
```diff
@@ -454,10 +417,11 @@ functions:
...
         script: |
           # Disable xtrace for security reasons (just in case it was accidentally set).
           set +x
+          . .evergreen/scripts/env.sh
```
Comment (Jibola) [inline .evergreen/config.yml:424]:
```suggestion
```

File: .evergreen/config.yml [hunk 16]
```diff
@@ -469,12 +433,13 @@ functions:
     - command: shell.exec
       type: test
       params:
-        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
+        include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```
Comment (Jibola) [inline .evergreen/config.yml:436]:
```suggestion
        include_expansions_in_env: ["DRIVERS_TOOLS", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
```

File: .evergreen/config.yml [hunk 17]
```diff
@@ -469,12 +433,13 @@ functions:
...
         working_dir: "src"
         script: |
           # Disable xtrace for security reasons (just in case it was accidentally set).
           set +x
           set -o errexit
+          . .evergreen/scripts/env.sh
```
Comment (Jibola) [inline .evergreen/config.yml:442]:
```suggestion
```

**Review by blink1073 — APPROVED** [review]

**Pull request merged**